### ISSUE TITLE
confuse about step 3 

### ISSUE THREAD

**Issue by tpys** [KaleidoZhouYN/Details-on-Face-Recognition#7]
Hi, I have tried first two step, the result seams good, FN does help converge a lot. But what do you mean by 
>   finetune lambda_min and lr 

Do i need to adjust lambda and lr to tuning, if so what reasonable value should i set, does they close to final stage: 

> lr = 1e-5 lambda_min = 0.5

**Comment by KaleidoZhouYN** [comment]
@tpys  emmm...I mean like original A-softmax,you need to  set a properly gamma and power so the lambda can get to the lamba_min slowly and this method won't make the converge crash.Because the parameter is hard to choose(always depend on the present converge status).I advice to change lambda and learning rate artificially until it reachs the final  status(lr = 1e-5 lambda_min =0.5)

**Comment by tpys** [comment]
@KaleidoZhouYN Thanks a lot. I kind of understand, in step 2 the lr = 0.1 lambd_min = 3, then i adjust them manually like lr = 1e-3 lambd_min = 2, then lr = 1e-4, lambda_min = 1. finally set lr = 1e-5, lambda_min = 0.5.  is that right?

**Comment by wuqiangch** [comment]
@tpys Does step2 resume training or finetune step1?

**Comment by tpys** [comment]
@wuqiangch resume training

**Comment by KaleidoZhouYN** [comment]
@tpys  yes,when constrain become stronger(lambda decrease),large lr may make the convergence crash,so you have to try different lr until it still looks like to converge.
by the way,extremely small lambda is also not good for Ms_celeb_1M dataset.

**Comment by tpys** [comment]
@KaleidoZhouYN Ok, I got it, thanks for you clear explanation.

**Comment by wuqiangch** [comment]
@tpys I have try step2 resume training from step1after step2, Acc drop from 94% to 74%. What wrong with this?

**Comment by KaleidoZhouYN** [comment]
@wuqiangch acc drop is a normal phenomenon when you change the m=1 to m=2.If you are confused about that,please think more about why sphereface works.

**Comment by philoxmyu** [comment]
@KaleidoZhouYN  what's the test accuracy of the final status (lr = 1e-5 lambda_min = 0.5 type = DOUBLE) . Thanks!

**Comment by Young1403** [comment]
@KaleidoZhouYN Hi,how much is your loss of the final status?Thanks!
